alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/modules/InstallModule.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2010-2022, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.server.modules
+
+import io.ktor.http.HttpMethod
+import io.ktor.serialization.kotlinx.json.json
+import io.ktor.server.application.Application
+import io.ktor.server.application.install
+import io.ktor.server.plugins.compression.Compression
+import io.ktor.server.plugins.compression.gzip
+import io.ktor.server.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.server.plugins.cors.routing.CORS
+
+/**
+ * Ktor module that adds all the [install] configuration to the application.
+ * The [io.ktor.serialization.kotlinx.json.JsonSupportKt.json] format is used for content negotiation.
+ * A cors configuration is included.
+ * Finally, an encoder for [Compression] is included.
+ * @see <a href="https://ktor.io/docs/modules.html">Ktor Modules</a>
+ */
+fun Application.installModule() {
+    install(ContentNegotiation) {
+        json()
+    }
+    install(CORS) {
+        allowMethod(HttpMethod.Get)
+        allowMethod(HttpMethod.Post)
+        anyHost()
+    }
+    install(Compression) {
+        gzip()
```

Comment:
Do I understand correctly that this means we send around gzip-compressed JSONs? Is there a quick handle for zstd instead?

Reply:
Yes, that is correct. The [compression algorithms](https://ktor.io/docs/compression.html) included in Ktor are just gzip and deflate. However, I think it may be possible to create a custom Compression Encoder and Decoder just by implementing some interface in the [Compression package](https://api.ktor.io/ktor-server/ktor-server-plugins/ktor-server-compression/io.ktor.server.plugins.compression/-compression.html).
Do I leave it like this for now?